chia/data_layer/data_store.py

```diff
@@ -1443,6 +1446,64 @@ async def remove_subscriptions(self, tree_id: bytes32, urls: List[str]) -> None:
                     },
                 )
 
+    async def delete_store_data(self, tree_id: bytes32) -> None:
+        async with self.db_wrapper.writer() as reader:
+            cursor = await reader.execute(
+                """
+                WITH all_nodes AS (
+                    SELECT a.hash, n.left, n.right
+                    FROM ancestors AS a
+                    JOIN node AS n ON a.hash = n.hash
+                    WHERE a.tree_id = ?
+                )
+
+                SELECT hash, left, right
+                FROM all_nodes
+                WHERE hash NOT IN (SELECT hash FROM ancestors WHERE tree_id != ?)
+                """,
+                (
+                    tree_id,
+                    tree_id,
+                ),
+            )
+            to_delete: Dict[bytes, Tuple[bytes, bytes]] = {}
+            ref_counts: Dict[bytes, int] = {}
+            async for row in cursor:
+                hash = row["hash"]
+                left = row["left"]
+                right = row["right"]
+                if hash in to_delete:
+                    prev_left, prev_right = to_delete[hash]
+                    assert prev_left == left
+                    assert prev_right == right
+                    continue
+                to_delete[hash] = (left, right)
+                if left is not None:
+                    ref_counts[left] = ref_counts.get(left, 0) + 1
+                if right is not None:
+                    ref_counts[right] = ref_counts.get(right, 0) + 1
+
+        async with self.db_wrapper.writer() as writer:
```

Comment:
With this being a separate db transactions, doesn't this leave open a race opportunity?  I think just getting the writer to begin with would be good even though we start with just reading.